storage/__init__.py

```diff
@@ -76,31 +76,36 @@ def regenerate(self):
         """Generate an auth token for the selected identity."""
         # If the server_url has a hash fragment, it's a storage node and
         # that's the secret.  Otherwise it's a token server url.
+        uid = self.uid
```

Comment:
Clearly we had never actually caused the `if url.fragment` codepath to be run before, because it was throwing on this missing local variable.